### PR TITLE
CrowdStrike Devices support in the device inventory

### PR DESCRIPTION
**Pull request by mahirchavda**

### CODE REVIEW

File: cyences_app_for_splunk/bin/cs_product_list.py [hunk 1]
```diff
@@ -231,6 +231,19 @@ def build_source_reviewer_search(by, values, first_call=True):
...
         ],
     },
+    {
+        "name": "CrowdStrike Devices",
+        "macro_configurations": [
+            {
+                "macro_name": "cs_crowdstrike_devices",
+                "label": "CrowdStrike Devices Data",
+                "search_by": "sourcetype",
+                "search_values": "crowdstrike:device:json",
```
Comment (hardikhdholariya) [inline cyences_app_for_splunk/bin/cs_product_list.py:241]: It seems this sourcetype has more info of devices compared to CrowdStrike:Event:Streams:JSON

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 2]
```diff
@@ -6106,24 +6106,25 @@ alert.track = 0
...
-| table time, indextime, product_name, product_uuid, crowdstrike_collected_by, crowdstrike_customer_id, ip, hostname, mac_address, user, crowdstrike_userid \
+| table _time, time, indextime,  product_name, product_uuid, hostname, user, ip, mac_address, crowdstrike_collected_by, external_ip, agent_version, site_name, status, first_seen, last_seen, last_login_timestamp, last_login_user, last_reboot, os_product_name, os_version, os_build, platform_name, product_type_desc \
 | cyencesdevicemanager operation="addentries" \
-| append [| inputlookup cs_crowdstrike_inventory ] \
-| stats latest(time) as time, latest(*) as * by crowdstrike_userid \
-| eval _key=crowdstrike_userid | outputlookup cs_crowdstrike_inventory
-action.cyences_notable_event_action.products = CrowdStrike EventStream
+| append [| inputlookup cs_crowdstrike_inventory] \
+| dedup product_uuid sortby -_time \
+| eval _key=product_uuid | outputlookup cs_crowdstrike_inventory
```
Comment (hardikhdholariya) [inline cyences_app_for_splunk/default/savedsearches.conf:6126]: we have to keep crowdstrike_userid field as it is displayed in device inventory raw expansion search.

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 3]
```diff
@@ -6106,24 +6106,25 @@ alert.track = 0
...
 display.general.type = statistics
 display.page.search.tab = statistics
 display.page.search.mode = fast
 request.ui_dispatch_app = cyences_app_for_splunk
 request.ui_dispatch_view = search
-search = `cs_crowdstrike_eventstream` | eval crowdstrike_userid=coalesce('event.UserId', 'event.UserName', user) | dedup crowdstrike_userid \
-| eval ip=lower(coalesce('event.UserIp', 'event.LocalIP', src)), hostname=lower(coalesce('event.ComputerName', 'event.HostnameField')), time=_time, indextime=_indextime, product_name="CrowdStrike", product_uuid=crowdstrike_userid \
-| rename host as crowdstrike_collected_by, "metadata.customerIDString" as crowdstrike_customer_id, "event.MACAddress" as mac_address \
+search = `cs_crowdstrike_devices` | dedup falcon_device.device_id \
+| eval time=_time, indextime=_indextime, product_name="CrowdStrike Devices" \
```
Comment (hardikhdholariya) [inline cyences_app_for_splunk/default/savedsearches.conf:6118]: For upgrade scenario, need to cleanup the device inventory lookup for product="CrowdStrike"

Also, on dashboard searches, need to update the product_names="CrowdStrike" to product_names="CrowdStrike Devices"

**Review by hardikhdholariya — CHANGES_REQUESTED** [review]

File: cyences_app_for_splunk/bin/upgrade_steps.py [hunk 4]
```diff
@@ -76,6 +76,22 @@ def upgrade_4_5_0(session_key, logger):
     handle_results(response, logger)
 
 
+def upgrade_4_8_0(session_key, logger):
+    service = client.connect(token=session_key, app=cs_utils.APP_NAME)
+
+    CROWDSTRIKE_DEVICES_CLEANUP_SEARCH = '| makeresults count=1 | eval time=now() | map search="| cyencesdevicemanager operation="cleanup" products_to_cleanup="CrowdStrike" minindextime=$time$"'
+    logger.info("Cleaning up the old Crowdstrike devices")
```
Comment (hardikhdholariya) [inline cyences_app_for_splunk/bin/upgrade_steps.py:83]: Add a reason to cleanup devices. same for the users cleanup log.

File: cyences_app_for_splunk/bin/upgrade_steps.py [hunk 5]
```diff
@@ -87,4 +103,5 @@ def upgrade_4_5_0(session_key, logger):
     ('4.3.0', upgrade_4_3_0),
     ('4.4.0', None),
     ('4.5.0', upgrade_4_5_0),
+    ('4.8.0', upgrade_4_8_0),
```
Comment (hardikhdholariya) [inline cyences_app_for_splunk/bin/upgrade_steps.py:106]: this change we are not going to include in this release, right?

Comment (mahirchavda) [reply cyences_app_for_splunk/bin/upgrade_steps.py:106]: yes. i think so

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 6]
```diff
@@ -6106,24 +6106,25 @@ alert.track = 0
...
 display.general.type = statistics
 display.page.search.tab = statistics
 display.page.search.mode = fast
 request.ui_dispatch_app = cyences_app_for_splunk
 request.ui_dispatch_view = search
-search = `cs_crowdstrike_eventstream` | eval crowdstrike_userid=coalesce('event.UserId', 'event.UserName', user) | dedup crowdstrike_userid \
-| eval ip=lower(coalesce('event.UserIp', 'event.LocalIP', src)), hostname=lower(coalesce('event.ComputerName', 'event.HostnameField')), time=_time, indextime=_indextime, product_name="CrowdStrike", product_uuid=crowdstrike_userid \
-| rename host as crowdstrike_collected_by, "metadata.customerIDString" as crowdstrike_customer_id, "event.MACAddress" as mac_address \
+search = `cs_crowdstrike_devices` | dedup falcon_device.device_id \
+| eval time=_time, indextime=_indextime, crowdstrike_id='falcon_device.device_id', product_name="CrowdStrike Devices" \
```
Comment (hardikhdholariya) [inline cyences_app_for_splunk/default/savedsearches.conf:6118]: I think let's keep product_name as is (**CrowdStrike**) then we don't have to change the on dashboard searches and it'll be consistent.

File: cyences_app_for_splunk/default/savedsearches.conf [hunk 7]
```diff
@@ -6106,24 +6106,25 @@ alert.track = 0
...
 | fillnull ip, hostname, mac_address, user value="" \
-| table time, indextime, product_name, product_uuid, crowdstrike_collected_by, crowdstrike_customer_id, ip, hostname, mac_address, user, crowdstrike_userid \
+| table _time, time, indextime,  product_name, product_uuid, hostname, user, ip, mac_address, crowdstrike_id, crowdstrike_collected_by, external_ip, agent_version, site_name, status, first_seen, last_seen, last_login_timestamp, last_login_user, last_reboot, os_product_name, os_version, os_build, platform_name, product_type_desc \
 | cyencesdevicemanager operation="addentries" \
-| append [| inputlookup cs_crowdstrike_inventory ] \
-| stats latest(time) as time, latest(*) as * by crowdstrike_userid \
-| eval _key=crowdstrike_userid | outputlookup cs_crowdstrike_inventory
-action.cyences_notable_event_action.products = CrowdStrike EventStream
+| append [| inputlookup cs_crowdstrike_inventory] \
+| dedup product_uuid sortby -_time \
```
[the remark this answers is not in the document]
Comment (mahirchavda) [reply cyences_app_for_splunk/default/savedsearches.conf]: Done

File: cyences_app_for_splunk/default/savedsearches.conf (hunk 6 above)
Comment (mahirchavda) [reply cyences_app_for_splunk/default/savedsearches.conf]: Done

**Review by hardikhdholariya — APPROVED** [review]

**Pull request merged**